### ISSUE TITLE
Problematic Vehicle Maintenance

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#360]
**Describe the bug**
- Too many vehicles are under maintenance
- Unclear who is performing the vehicle maintenance 

**Expected behavior**
- The status should be changed back to `Parked` when the maintenance is done. 
- The Vehicle Window's Maint tab should a decent a maintenance (history) log with the following data : 
1. when it starts, 
2. what's involved, 
3. when it ends.
4. who does what.
 
**Screenshots**
Too many vehicle under maintenance : 
![image](https://user-images.githubusercontent.com/1168584/121437201-c24a5380-c936-11eb-9f23-5d9e6feeadcc.png)

No one is doing maintenance : 
![image](https://user-images.githubusercontent.com/1168584/121437329-f756a600-c936-11eb-972b-c25466dffa1e.png)

Mobile B from TeslaVille has NEVER been on any mission. Yet it was put on maintenance since when ?
![image](https://user-images.githubusercontent.com/1168584/121437529-5ae0d380-c937-11eb-9011-ea89b0d8f423.png)

**Desktop (please complete the following information):**
 - Build 5790

**Comment by mokun** [comment]
See this console printout regarding the vehicles in Teslaville : 

```
Teslaville >  ​/v
 Summary
 --------------------------------------------
          Total # of Rovers : 6
    Cargo Rovers on Mission : 0
      Transports on Mission : 0
       Explorers on Mission : 0
            LUVs on Mission : 0
          Rovers on Mission : 0
Parked/Garaged Cargo Rovers : 0
  Parked/Garaged Transports : 0
   Parked/Garaged Explorers : 0
        Parked/Garaged LUVs : 2
      Rovers NOT on mission : 6
 Vehicles
 --------------------------------------------
                  Name |            Type |                   Mission |                   Lead
---------------------------------------------------------------------------------------------
            Desert Fox |     Cargo Rover |                           |                       
              Mobile B |  Explorer Rover |                           |                       
                Gypsea |  Explorer Rover |                           |                       
               LUV 004 |             LUV |                           |                       
               LUV 005 |             LUV |                           |                       
              Mobile A | Transport Rover |                           |                       

```

**Comment by mokun** [comment]
I thought the switch `/sp` is `specs` but is actually used by `speed`  

> Mobile B >  ​/i
> Inventory held
>                    Capacity : 2500.00 kg
>                   Available : 2500.00 kg
>                           Item |     Amount
> -------------------------------------------
> 
> Mobile B >  ​/ai
>                       Name |          State | Active |               Operator | Use | Inner | Outer
> ---------------------------------------------------------------------------------------------------
>                   Mobile B |    PRESSURIZED |     No |                   None | 0/2 |   ULK |   LCK
> 
> Mobile B >  ​/ml
> 
> Mobile B >  ​/m
> No mission
> Mobile B >  ​/sp
> The target simulation ratio is x128, speed 7
> Change (Y/N)? ​n
> Mobile B >  ​

**Comment by bevans2000** [comment]
> I thought the switch `/sp` is `specs` but is actually used by `speed`

Yes it's a bug. There is an overlap of the 2 short commands. Just change either of them.

**Comment by bevans2000** [comment]
I noticed that there did seems to be a lot of people warning there could not enter the Airlock of Vehicles and I think that is why the maintenance is not getting completed.

**Comment by mokun** [comment]
No one seems to be working on maintaining these vehicles.

If they come back from a mission, I can see that the need for maintenance is there. But if they just park it there, other than periodic maintenance and test, then the wear and tear condition should not be 99% or 1000% and should be lower, until some work is done.

![image](https://user-images.githubusercontent.com/1168584/123204992-f6ad2c00-d46d-11eb-8b5e-324d74d3708f.png)

**Comment by mokun** [comment]
I take that back. 

They are indeed working on these 3 vehicles
 
![image](https://user-images.githubusercontent.com/1168584/123205621-198c1000-d46f-11eb-9087-653b483e471b.png)

However, interestingly, these 3 vehicles under maintenance never went out on mission at all. 

On the other hand, the 7 vehicles that did go out for excursions never need to be worked on. 

![image](https://user-images.githubusercontent.com/1168584/123205787-5eb04200-d46f-11eb-8b04-a01b7d743bdb.png)

Seems we need to model how a vehicle came back from a mission in need of some work.

**Comment by mokun** [comment]
@bevans2000 , 

What do you think we should do for vehicles that just come back from a mission and how ?

**Comment by bevans2000** [comment]
I think a Vehicle returning from a Mission should be treated no differently. Maintenance task should be trigger by the stats on the Vehicle. Sure a Mission should add more "age" to the Vehicle but I don't see a need for an explicit maintenance at the end of a Mission.
I think the correct thing is to limit how many Vehicles can be reserved for Maintenance at a Settlement and make sure the most needy are at the front of the queue. So it would be possible that "used" Vehicles and assigned to new Missions. Also maybe only reserve a Vehicle once someone actually starts on the Maintenance.

**Comment by mokun** [comment]
> I think a Vehicle returning from a Mission should be treated no differently. Maintenance task should be trigger by the stats on the Vehicle. 

We could have vehicles going through a different checkpoint inspections, depending on its history and its type.

On the flipside, this makes me think that if a vehicle is on mission, compared to a vehicle that stay in a settlement, in real world, certain subsystems are put into heavy use, such as the drivetrain. 

In that case, the malfunction should be higher when the vehicle is being driven. 

But currently, I don't our malfunction modeling capture this reality.

Supposedly, when a vehicle is moving, it calls *MalfunctionManager's activeTimePassing()*

**Comment by mokun** [comment]
> Sure a Mission should add more "age" to the Vehicle but I don't see a need for an explicit maintenance at the end of a Mission.

A vehicle's maintenance log should be recorded. 

Can we further model the vehicle in such a way that the reliability of the subsystems is tracked ? 

Say vehicle A has recurrent engine issues.

At some point, rather than replacing parts in the engine, they may decide to replace the whole engine because it has higher probability of failure

**Comment by mokun** [comment]
@bevans2000 ,

btw, we already have `odometerMileage` in Vehicle class. But it's not being used for anything practical.
